nb_train.py

```diff
@@ -4,87 +4,159 @@
-    all_words = nltk.FreqDist(w.lower() for w in words_flat) 
-    comm_words = list(all_words)[:x]
+        Args:
+        data(list): a list of words
+        count(int): the number of most common words to be extracted.'''
+       
+    word_frequency = nltk.FreqDist(w.lower() for w in data) 
+    common_words = list(word_frequency)[:count]
     
-    return comm_words
+    return common_words
 
-def review_features(review):
-    '''decompose a review into unique words, returns a dict mapping features to booleans if the feature is in a given review'''
+def find_features(word_list, feature_words):
+    '''Select unique words from a list of words, returns a dictionary mapping features to booleans.
+ 
+        Args:
+        word_list(list): a list of words split from a sentence
+        feature_words(list): a list of common words
+        
+        Returns:
+        features(dict): a dictionary with each word in feature_words as keys and booleans as values 
+        (T if the word appears in word_list, F if otherwise).'''
     
-    words = set(review) 
+    words = set(word_list)
     features = {}
-    for w in commwords:
+    for w in feature_words:  
         features[f'contains({w})'] = (w in words)
         
     return features
 
 def train_model(data):
-    '''Train the model, return the model'''
+    '''Trains the model, returns the model.
     
-    classifier = nltk.NaiveBayesClassifier.train(data)   
+        Args:
+        data(list): a list of tuples where the first element of each tuple is a dict of features, the second element is the sentiment.'''
+    
+    classifier = nltk.NaiveBayesClassifier.train(data)  
     
     return classifier
 
 
-#Pair review to corresponding sentiment
-review_to_sentiment = rev_sent(smalldf)
-
-#Extract 2000 most common words
-commwords = commwords_extractor(review_to_sentiment, 2000)
-
 def main():
     
-    featuresets = [(review_features(rev), label) for (rev, label) in review_to_sentiment]
-    train_set, test_set = featuresets[:1500], featuresets[1500:]
+    with ZipFile('imdb ratings.zip') as zf:
+        f = zf.open(DATA)
+    
+    df = pd.read_csv(f)
+    smalldf = df[:DATA_LENGTH]
+    smalldf = smalldf.drop_duplicates(ignore_index = True)
+    smalldf.loc[smalldf[LABEL] == LABEL_0, LABEL] = NEGATIVE_SENTIMENT
+    smalldf.loc[smalldf[LABEL] == LABEL_1, LABEL] = POSITIVE_SENTIMENT
+    
+    
+    # Pair each string to its corresponding sentiment.
+    str_to_sentiment = make_text_label_tuples(smalldf)  
+    
+    random.shuffle(str_to_sentiment) # randomly shuffle the list of tuples to randomize the contents of train and test data.
+    
+    nested_words = [t[0] for t in str_to_sentiment] 
+    
+    words_flatlist = [word for wl in nested_words for word in wl]
+    
+    common_words = extract_common_words(words_flatlist, 2000)
+    
+    # TODO: should I only find features from the train data? Am I overfitting by also finding features from the test set?
```

Comment:
You should only train on the training data. If you train on the test data, your model will learn the specific characteristics of the test data, which means that any accuracy you get by testing on it will be unrepresentative